cluster/repository/cluster_repository_blackbox_test.go

```diff
@@ -148,43 +161,97 @@ func (s *clusterTestSuite) TestSaveOK() {
 	require.NoError(s.T(), err)
 	test.AssertEqualClusters(s.T(), cluster2, loaded2)
 }
 
 func (s *clusterTestSuite) TestSaveUnknownFails() {
+	// given
 	id := uuid.NewV4()
+	// when
 	err := s.repo.Save(context.Background(), &repository.Cluster{ClusterID: id})
+	// then
 	test.AssertError(s.T(), err, errors.NotFoundError{}, "cluster with id '%s' not found", id)
 }
 
 func (s *clusterTestSuite) TestExists() {
+	// given
 	id := uuid.NewV4()
 	err := s.repo.CheckExists(context.Background(), id.String())
 	test.AssertError(s.T(), err, errors.NotFoundError{}, "cluster with id '%s' not found", id)
-
+	// when
 	cluster := test.CreateCluster(s.T(), s.DB)
+	// then
 	err = s.repo.CheckExists(context.Background(), cluster.ClusterID.String())
 	require.NoError(s.T(), err)
 }
 
 func (s *clusterTestSuite) TestQueryOK() {
+	// given
 	cluster1 := test.CreateCluster(s.T(), s.DB)
-
+	// when
 	clusters, err := s.repo.Query(func(db *gorm.DB) *gorm.DB {
 		return db.Where("cluster_id = ?", cluster1.ClusterID)
 	})
-
+	// then
 	require.NoError(s.T(), err)
 	require.Len(s.T(), clusters, 1)
 	test.AssertEqualClusters(s.T(), cluster1, &clusters[0])
 }
+
+func (s *clusterTestSuite) TestList() {
+	// given
+	cluster1 := &repository.Cluster{
+		AppDNS:            uuid.NewV4().String(),
+		AuthClientID:      uuid.NewV4().String(),
+		AuthClientSecret:  uuid.NewV4().String(),
+		AuthDefaultScope:  uuid.NewV4().String(),
+		ConsoleURL:        uuid.NewV4().String(),
+		LoggingURL:        uuid.NewV4().String(),
+		MetricsURL:        uuid.NewV4().String(),
+		Name:              uuid.NewV4().String(),
+		SAToken:           uuid.NewV4().String(),
+		SAUsername:        uuid.NewV4().String(),
+		SATokenEncrypted:  true,
+		TokenProviderID:   uuid.NewV4().String(),
+		Type:              "bar",
+		URL:               "http://" + uuid.NewV4().String() + "/",
+		CapacityExhausted: false,
+	}
+	err := s.repo.Create(context.Background(), cluster1)
+	require.NoError(s.T(), err)
+	cluster2 := &repository.Cluster{
+		AppDNS:            uuid.NewV4().String(),
+		AuthClientID:      uuid.NewV4().String(),
+		AuthClientSecret:  uuid.NewV4().String(),
+		AuthDefaultScope:  uuid.NewV4().String(),
+		ConsoleURL:        uuid.NewV4().String(),
+		LoggingURL:        uuid.NewV4().String(),
+		MetricsURL:        uuid.NewV4().String(),
+		Name:              uuid.NewV4().String(),
+		SAToken:           uuid.NewV4().String(),
+		SAUsername:        uuid.NewV4().String(),
+		SATokenEncrypted:  true,
+		TokenProviderID:   uuid.NewV4().String(),
+		Type:              "bar",
+		URL:               "http://" + uuid.NewV4().String() + "/",
+		CapacityExhausted: false,
+	}
+	err = s.repo.Create(context.Background(), cluster2)
+	require.NoError(s.T(), err)
+	// when
+	clusters, err := s.repo.List(context.Background())
+	// then
+	require.NoError(s.T(), err)
+	require.Len(s.T(), clusters, 2)
+	test.AssertEqualClusters(s.T(), cluster1, &clusters[0])
```

Comment:
You do not order clusters when query them in DB, so, theoretically they can be in a different order. Worth adding `test.AssertClusters(t, expected, actual []Cluster)` or `test.AssertClusters(t, actual []Cluster, expected ...Cluster)` since it can be useful in other tests too.

Reply:
good point, let me fix that

Reply:
done in dc31c0f